src/test/java/org/apache/commons/lang3/ArrayUtilsBinarySearchTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.lang3;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Unit tests {@link ArrayUtils} binarySearch functions.
+ */
+public class ArrayUtilsBinarySearchTest extends AbstractLangTest {
+
+    @Test
+    public void binarySearch_whenEmpty_returnM1() {
```

Comment:
Since all tests in the class call `binarySearch()` and the class is already called "binary search", you do not need to repeat that name in each test method name, that'll de-clutter names a bit.

Reply:
I don't agree with this comment.  The methods names are intentionally descriptive with pattern `name-condition-expected`.  This improves readability.  While there are truly only `binarySearch` now, it doesn't have to be forever and vice versa - this class may be possibly merged into more generic ones which would work with sorted arrays (which was the very original name).